src/nodes/utils.ts

```diff
@@ -310,6 +313,14 @@ function generateRandomNodeIdForBucket(
   return xOrNodeId(nodeId, randomDistanceForBucket);
 }
 
+function isConnectionError(e): boolean {
```

Comment:
Need a comment here what this is for. What constitutes a connection error and why is it being checked here?